Userland/Libraries/LibWeb/Fetch/Fetching/Fetching.cpp

```diff
@@ -708,8 +710,83 @@ WebIDL::ExceptionOr<JS::NonnullGCPtr<PendingResponse>> scheme_fetch(JS::Realm& r
     }
     // -> "blob"
     else if (request->current_url().scheme() == "blob"sv) {
-        // FIXME: Support 'blob://' URLs
-        return PendingResponse::create(vm, request, Infrastructure::Response::network_error(vm, "Request has 'blob:' URL which is currently unsupported"sv));
+        auto const& store = FileAPI::blob_url_store();
+
+        // 1. Let blobURLEntry be request’s current URL’s blob URL entry.
+        auto blob_url_entry = store.get(TRY_OR_THROW_OOM(vm, request->current_url().to_string()));
+
+        // 2. If request’s method is not `GET`, blobURLEntry is null, or blobURLEntry’s object is not a Blob object,
+        //    then return a network error. [FILEAPI]
+        if (request->method() != "GET"sv.bytes() || !blob_url_entry.has_value()) {
+            // FIXME: Handle "blobURLEntry’s object is not a Blob object". It could be a MediaSource object, but we
+            //        have not yet implemented the Media Source Extensions spec.
+            return PendingResponse::create(vm, request, Infrastructure::Response::network_error(vm, "Request has an invalid 'blob:' URL"sv));
+        }
+
+        // 3. Let blob be blobURLEntry’s object.
+        auto const& blob = blob_url_entry->object;
+
+        // 4. Let response be a new response.
+        auto response = Infrastructure::Response::create(vm);
+
+        // 5. Let fullLength be blob’s size.
+        auto full_length = blob->size();
+
+        // 6. Let serializedFullLength be fullLength, serialized and isomorphic encoded.
+        auto serialized_full_length = TRY_OR_THROW_OOM(vm, String::number(full_length));
+
+        // 7. Let type be blob’s type.
+        auto const& type = blob->type();
+
+        // 8. If request’s header list does not contain `Range`:
+        if (!request->header_list()->contains("Range"sv.bytes())) {
+            // 1. Let bodyWithType be the result of safely extracting blob.
+            auto body_with_type = TRY(safely_extract_body(realm, blob));
+
+            // 2. Set response’s status message to `OK`.
+            response->set_status_message(TRY_OR_THROW_OOM(vm, ByteBuffer::copy("OK"sv.bytes())));
```

Comment:
ByteBuffer has an inline capacity, this cannot OOM - see also:

![image](https://github.com/SerenityOS/serenity/assets/19366641/93763a76-c632-4890-acf2-76ad52b84c40)